evmap/src/read.rs

```diff
@@ -0,0 +1,257 @@
+    /// refreshed by the writer. If no refresh has happened, or the map has been destroyed, this
+    /// function returns `None`.
+    #[inline]
+    pub fn get_one<'rh, Q: ?Sized>(&'rh self, key: &'_ Q) -> Option<ReadGuard<'rh, V>>
+    where
+        K: Borrow<Q>,
+        Q: Hash + Eq,
+    {
+        ReadGuard::try_map(self.get_raw(key.borrow())?, |x| x.get_one())
+    }
+
+    /// Returns a guarded reference to the values corresponding to the key along with the map
+    /// meta.
+    ///
+    /// While the guard lives, the map cannot be refreshed.
+    ///
+    /// The key may be any borrowed form of the map's key type, but `Hash` and `Eq` on the borrowed
+    /// form *must* match those for the key type.
+    ///
+    /// Note that not all writes will be included with this read -- only those that have been
+    /// refreshed by the writer. If no refresh has happened, or the map has been destroyed, this
+    /// function returns `None`.
+    ///
+    /// If no values exist for the given key, `Some(None, _)` is returned.
+    pub fn meta_get<Q: ?Sized>(&self, key: &Q) -> Option<(Option<ReadGuard<'_, Values<V, S>>>, M)>
+    where
+        K: Borrow<Q>,
+        Q: Hash + Eq,
+    {
+        let inner = self.handle.enter()?;
+        if !inner.ready {
+            return None;
+        }
+        let meta = inner.meta.clone();
+        let res = ReadGuard::try_map(inner, |inner| inner.data.get(key));
+        Some((res, meta))
+    }
+
+    /// Returns true if the writer has destroyed this map.
+    ///
+    /// See [`WriteHandle::destroy`].
+    pub fn is_destroyed(&self) -> bool {
+        self.handle.is_destroyed()
+    }
+
+    /// Returns true if the map contains any values for the specified key.
+    ///
+    /// The key may be any borrowed form of the map's key type, but `Hash` and `Eq` on the borrowed
+    /// form *must* match those for the key type.
+    pub fn contains_key<Q: ?Sized>(&self, key: &Q) -> bool
+    where
+        K: Borrow<Q>,
+        Q: Hash + Eq,
+    {
+        self.enter().map_or(false, |x| x.contains_key(key))
+    }
+
+    /// Returns true if the map contains the specified value for the specified key.
+    ///
+    /// The key and value may be any borrowed form of the map's respective types, but `Hash` and
+    /// `Eq` on the borrowed form *must* match.
+    pub fn contains_value<Q: ?Sized, W: ?Sized>(&self, key: &Q, value: &W) -> bool
+    where
+        K: Borrow<Q>,
+        V: Borrow<W>,
+        Q: Hash + Eq,
+        W: Hash + Eq,
+    {
+        self.get_raw(key.borrow())
+            .map(|x| x.contains(value))
+            .unwrap_or(false)
+    }
+
+    /// Read all values in the map, and transform them into a new collection.
+    pub fn map_into<Map, Collector, Target>(&self, mut f: Map) -> Collector
+    where
+        Map: FnMut(&K, &Values<V, S>) -> Target,
+        Collector: FromIterator<Target>,
+    {
+        Collector::from_iter(self.enter().iter().flatten().map(|(k, v)| f(k, v)))
```

Comment:
Clippy suggesting replacing from_iter -> collect
```suggestion
        self.enter()
            .iter()
            .flatten()
            .map(|(k, v)| f(k, v))
            .collect()
```